lms/djangoapps/grades/models.py

```diff
@@ -236,9 +241,43 @@ def bulk_get_or_create(cls, block_record_lists, course_key):
         BlockRecordList objects for the given course_key, but
         only for those that aren't already created.
         """
-        existent_records = {record.hashed: record for record in cls.bulk_read(course_key)}
+        existent_records = cls.bulk_read(course_key)
         non_existent_brls = {brl for brl in block_record_lists if brl.hash_value not in existent_records}
-        cls.bulk_create(non_existent_brls)
+        created = cls.bulk_create(non_existent_brls)
+        cls.update_cache(course_key, created)
+
+
+    @classmethod
+    def prefetch(cls, course_key):
```

Comment:
Consider renaming this method so it's consistent with the other cache related methods.  For example: `get_from_cache`.  So, the methods would be:
```
_get_from_cache
_update_cache
clear_cache
```